coding-guidelines/core/feature-flags.md

```diff
@@ -0,0 +1,196 @@
+## Introduction
+Feature flags enable the developer to create new code which is hidden behind the flag and merge it into the trunk branch, even when the code is not finalized.
+We use this functionality to merge breaks into the trunk early, without them already being switched active. To learn more about breaking changes and backward compability take a look to our [Backward Compatibility Guide](/Product/Guides/Development/BackwardCompatibility)
+
+Meanwhile we have only one feature flag in our core sources: `v6.5.0.0`. This feature flag is used for the above mentioned breaks.
+
+### Activating the flag
+To switch flags on and off you can use the ***.env*** to configure each feature flag. Using dots inside an env variable are not allowed, so we use underscore instead:
+```
+V6_5_0_0=1
+```
+
+## Using flags in PHP
+The feature flag can be used in PHP to make specific code parts only executable when the flag is active.
+
+### Using flags in methods
+When there is no option via the container you can use additional helper functions:
+```php
+use Shopware\Core\Framework\Feature;
+ 
+class ApiController
+{
+
+  public function indexAction(Request $request)
+  {
+    // some old stuff
+    Feature::ifActiveCall('v6.5.0.0', $this, 'handleNewFeature', $request);
+    // some old stuff
+  }
+
+  private function handleNewFeature(Request $request)
+  {
+    // awesome new stuff
+  }
+}
+```
+
+You can also do it in a callback:
+```php
+use Shopware\Core\Framework\Feature;
+ 
+class ApiController
+{
+  public function indexAction(Request $request)
+  {
+    // some old stuff
+    Feature::ifActive('v6.5.0.0', function() use ($request) {
+      // awesome stuff
+    });
+    // some old stuff
+  }
+}
+```
+
+And you can use it for conditions:
+```php
+use Shopware\Core\Framework\Feature;
+ 
+class ApiController
+{
+  public function indexAction(Request $request)
+  {
+    // some old stuff
+    if (Feature::isActive('v6.5.0.0')) {
```

Comment:
we should document it the other way around, as this would be easier to remove with the next major